eureka/S2_calibrations/s2_calibrate.py

```diff
@@ -65,16 +66,34 @@ def calibrateJWST(eventlabel, ecf_path='./', s1_meta=None):
     meta = readECF.MetaClass(ecf_path, ecffile)
     meta.eventlabel = eventlabel
 
-    # Create directories for Stage 2 processing outputs
-    meta.inputdir_raw = meta.inputdir
-    meta.outputdir_raw = meta.outputdir
+    if s1_meta == None:
+        #load savefile
+        s1_meta = read_s1_meta(meta)
+
+    if s1_meta != None:
+        meta = load_general_s1_meta_info(meta, ecf_path, s1_meta)
+    else:
+        # Attempt to find subdirectory containing S1 FITS files
+        meta = find_s1_files(meta)
+
+        # Create directories for Stage 2 processing outputs
+        meta.inputdir_raw = meta.inputdir
+        meta.outputdir_raw = meta.outputdir
+        meta.inputdir = os.path.join(meta.topdir, *meta.inputdir_raw.split(os.sep))
```

Comment:
It would probably be cleaner to never join topdir and inputdir/outputdir, so that they don't need to be split later. Instead, we should combine the two strings only for the actual read/write step.

Reply:
The issue is that we then need to duplicate the os.path.join code tens of times for every read/write function and plotting function. My idea had been to have inputdir_raw and outputdir_raw be inputdir and outputdir without the topdir joined on so that we still have them if needed.

Reply:
Ok, I'm fine with going that route but I suggest we rename our variables to avoid confusion.  For example, define meta.s3_topinputdir = os.path.join(meta.s3_topdir, meta.s3_inputdir).  Let me know if I'm wrong here, but I don't think there's any reason to be overwriting meta.inputdir at every stage.  Not sure if having a different topdir for each stage is useful, so that's something we can discuss.

Reply:
Adding to my last comment, this may mean passing `meta.s3_topinputdir` as a variable to `readECF.py`.

Reply:
Are you fine with leaving this as I've written it for now and then this will all get significatly overhauled in a few days?